tests/apps/test_data.py

```diff
@@ -328,3 +332,111 @@ def test_data_dim_correct(self, dataset_patched):
         assert m == dataset_patched.n_features
         assert p == len(dataset_patched.adjs)
         assert q == len(dataset_patched.unit_data)
+
+
+@pytest.mark.parametrize("datasets", WATER_DATASETS_LIST)
+class TestWaterDatasets:
+    """Tests for the ``Water`` class"""
+
+    @pytest.fixture
+    def dataset(self, datasets):
+        """Fixture for loading each of the datasets in ``WATER_DATASETS_LIST``"""
+        yield datasets(t=0)
+
+    def test_w_dims(self, dataset):
+        """Test if w has correct shape"""
+        w = dataset.w
+        assert w.shape == (dataset.modes,)
+
+    def test_u_dims(self, dataset):
+        """Test if U has correct shape"""
+        U = dataset.U
+        assert U.shape == (dataset.modes, dataset.modes)
+
+    def test_u_unitary(self, dataset):
+        """Test if U is unitary"""
+        U = dataset.U
+        assert np.allclose(np.eye(U.shape[0]), U.conj().T @ U)
+
+    def test_sample_dims(self, dataset):
+        """Test if sample has correct shape"""
+        assert np.shape(dataset[:]) == (5000, dataset.modes)

```

Comment:
I feel like this one should be looped over the available times:
```suggestion
    @pytest.mark.parametrize("times", datasets.available_times)
    def test_sample_dims(self, times):
        """Test if sample has correct shape"""
        dataset = datatests(times)
        assert np.shape(dataset[:]) == (5000, dataset.modes)
```